brave/brave-core
# Fixed extensions import from Chromium based browsers

Resolves https://github.com/brave/brave-browser/issues/27770

The issue is that Chrome removed extensions list from `Secure Preferences` but some of the browsers still using it. Added extensions list fetching from both places `Secure Preferences` and `Preferences` files to support both ways to keep compatibility with all Chromium based browsers. 
Additionally refactored the import utils to remove existing code duplication to fetch extensions list from external importer and importer list. It allowed to add more tests to import utils.

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Install latest Chrome (for win I use 109.0.5414.75 (Official Build) (64-bit)) and install an extension from web store, I use jfbnmfgkohlfclfnplnlenbalpppohkm for testing
- Check `c:\Users\[username]\AppData\Local\Google\Chrome\User Data\Default\Secure Preferences` file, it should have one line without extensions info.
- Try to import from Chrome profile
- Make sure installed extension imported to Brave
- Additionally check extensions can be imported from Chrome and others Chromium based browsers who keeps information inside `Secure Preferences`, I checked with Yandex.
